### PR TITLE
[0.35] Deployment rollbacks

### PR DESCRIPTION
**Pull request by jwitz**

### CODE REVIEW

File: software/deploy-rollbacks.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+description: Learn how to roll back to a deploy, which lets you run earlier version of your project code.
+---
+
+Deploy rollbacks are an emergency option if a Deployment unexpectedly stops working after a recent deploy. For example, if one of your DAGs worked in development but suddenly fails in a mission-critical production Deployment, you can roll back to your previous deploy to quickly get your pipeline running again. This allows you to troubleshoot the issue more thoroughly in development before redeploying to production. You can roll back to any deploy in the last three months regardless of your Runtime version, DAG code, or Deployment settings.
+
+## Prerequisites
+
+- You must be at least a Deployment Admin or otherwise have permissions to upgrade a Deployment.
+- Your Deployment must be on Astro Runtime 5 (Airflow 2.3) or later. Rolling back to any version before Astro Runtime 5 is not supported. 
+- Your Deployment must be configured to use standard image and DAG-only deploys. Rollbacks are not supported for NFS deploys and Git sync deploys.
```
Comment (jwitz) [inline software/deploy-rollbacks.md:14]: I made this assumption

File: software/deploy-rollbacks.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+## Enable deploy rollbacks
+
+1. In the Software UI, open your Deployment.
+2. On the **Settings** page, toggle **Rollback Deploy** to **Enable**.
+
+## Roll back to a deploy
+
+:::danger
+
+Astronomer recommends triggering Deployment rollbacks only as a last resort for recent deploys that aren't working as expected. Deployment rollbacks can be disruptive, especially if you triggered multiple deploys between your current version and the rollback version. See [What happens during a deploy rollback](#what-happens-during-a-deploy-rollback) before you trigger a rollback to anticipate any unexpected effects.
```
Comment (jwitz) [inline software/deploy-rollbacks.md:25]: Warning copied from Astro docs, I assume the same applies in Software?

File: software/deploy-rollbacks.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
+---
+sidebar_label: 'Roll back a deploy'
+title: 'Roll back an image deploy on Astronomer Software'
+id: deploy-rollbacks
+description: Learn how to roll back to a deploy, which lets you run earlier version of your project code.
+---
+
+Deploy rollbacks are an emergency option if a Deployment unexpectedly stops working after a recent deploy. For example, if one of your DAGs worked in development but suddenly fails in a mission-critical production Deployment, you can roll back to your previous deploy to quickly get your pipeline running again. This allows you to troubleshoot the issue more thoroughly in development before redeploying to production. You can roll back to any deploy in the last three months regardless of your Runtime version, DAG code, or Deployment settings.
```
Comment (danielhoherd) [inline software/deploy-rollbacks.md:8]:
```suggestion
Deploy rollbacks are an emergency option if a Deployment unexpectedly stops working after a recent deploy. For example, if one of your DAGs worked in development but suddenly fails in a mission-critical production Deployment, you can roll back to your previous deploy to quickly get your pipeline running again. This allows you to revert your deployment to a known good state while you investigate the cause of the failure. You can roll back to any deploy in the last three months regardless of your Runtime version, DAG code, or Deployment settings.
```

> You can roll back to any deploy in the last three months

Is this true? CC @tonyHuinker @rishkarajgi @vishwas-astro

File: software/deploy-rollbacks.md (hunk 1 above)
Comment (danielhoherd) [inline software/deploy-rollbacks.md:14]:
```suggestion
- Your Deployment must be configured to use image or DAG-only deploys. Rollbacks are not supported for NFS deploys and Git sync deploys.
```

**Review by danielhoherd — APPROVED** [review]
Seems good to me. I left some optional suggestions and a question about accuracy. I'd like to see others review the changes too though.

File: software/deploy-rollbacks.md (hunk 2 above)
Comment (Simpcyclassy) [reply software/deploy-rollbacks.md:25]: Yes.

**Comment by lzdanski** [comment]
Confirmed with Daniel H the docs still look good.

File: software/deploy-rollbacks.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+3. Write a description for why you're triggering the rollback, then confirm the rollback.
+
+### What happens during a deploy rollback
+
+A deploy rollback is a new deploy of a previous version of your code. This means that the rollback deploy appears as a new deploy in **Deploy History**, and the records for any deploys between your current version and rollback version are still preserved. In Git terms, this is equivalent to `git revert`.
+
+When you trigger a rollback, the following information is rolled back:
+
+- All project code, including DAGs.
+- Your Astro Runtime version. Note that there are no limitations to rolling back to restricted Runtime versions, which can include major bugs and performance issues.
```
Comment (lzdanski) [inline software/deploy-rollbacks.md:41]:
```suggestion
- Your Astro Runtime version. 
```

File: software/deploy-rollbacks.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+- Your Deployment's DAG deploy setting.
+
+The following information isn't rolled back:
+
+- Your Deployment's resource configurations, such as executor and scheduler configurations.
+- Your Deployment's environment variable values.
+- Any other Deployment settings that you configure through the Astro UI, such as your Deployment name and description. 
+- For Runtime version downgrades, any data related to features that are not available in the rollback version are erased from the metadata database and not recoverable.
+
+Logs related to the rollback are exported to Elasticsearch.
```
Comment (lzdanski) [inline software/deploy-rollbacks.md:51]:
```suggestion
    :::danger Airgapped Environments Only

    Some versions of the Astro runtime are restricted, and you cannot rollback to or create new Deployments with these restricted versions. However, if you work in an Airgapped environment, the JSON list of available Astro Runtime versions must be [manually updated](software/install-airgapped.md#step-5-fetch-airflow-updates). Depending on the cadence that your organization updates the list of available Runtime versions, you might be able to roll back to a restricted Astro Runtime version, which can include major bugs and performance issues.

    :::


Logs related to the rollback are exported to Elasticsearch.
```

File: software/deploy-rollbacks.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+3. Write a description for why you're triggering the rollback, then confirm the rollback.
+
+### What happens during a deploy rollback
+
+A deploy rollback is a new deploy of a previous version of your code. This means that the rollback deploy appears as a new deploy in **Deploy History**, and the records for any deploys between your current version and rollback version are still preserved. In Git terms, this is equivalent to `git revert`.
+
+When you trigger a rollback, the following information is rolled back:
+
+- All project code, including DAGs.
+- Your Astro Runtime version. Note that you cannot roll back to a restricted Runtime version, as they can include major bugs and performance issues.
```
Comment (lzdanski) [inline software/deploy-rollbacks.md:41]:
```suggestion
- Your Astro Runtime version. Note that if you roll back to a restricted Runtime version they can include major bugs and performance issues.
```

File: software/deploy-rollbacks.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+- Your Deployment's environment variable values.
+- Any other Deployment settings that you configure through the Astro UI, such as your Deployment name and description. 
+- For Runtime version downgrades, any data related to features that are not available in the rollback version are erased from the metadata database and not recoverable.
+
+    :::danger Airgapped Environments Only
+
+    Some versions of the Astro runtime are restricted, and you cannot rollback to or create new Deployments with these restricted versions. However, if you work in an Airgapped environment, the JSON list of available Astro Runtime versions must be [manually updated](software/install-airgapped.md#step-5-fetch-airflow-updates). Depending on the cadence that your organization updates the list of available Runtime versions, you might be able to roll back to a restricted Astro Runtime version, which can include major bugs and performance issues.
+
+    :::
+
```
Comment (lzdanski) [inline software/deploy-rollbacks.md:56]:
```suggestion
```

**Pull request merged**